e2e/async-esm-import/__tests__/test.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import {jest} from '@jest/globals';
+
+test('should have a fresh module state in each isolateModulesAsync context', async () => {
+  await jest.isolateModulesAsync(async () => {
+    const {getState, incState} = await import('../main.js');
```

Comment:
Also rename the actual file to `main.mjs`.

```suggestion
    const {getState, incState} = await import('../main.mjs');
```